src/main/scala/cromwell/engine/backend/local/LocalBackend.scala

```diff
@@ -75,4 +97,71 @@ class LocalBackend extends Backend {
+    hostExecutionDirectory.mkdirs()
+
+    val hostExecutionAbsolutePath = hostExecutionDirectory.getAbsolutePath
+    val hostInputsPath = Paths.get(hostExecutionAbsolutePath, "workflow-inputs")
+    hostInputsPath.toFile.mkdir()
+    val hostOutputsPath = Paths.get(hostExecutionAbsolutePath, "workflow-outputs")
+    hostOutputsPath.toFile.mkdir()
+    stageAndLocalizeWorkflowInputs(workflow, actualInputs)
+  }
+
+  /**
+   * {{{cromwell-executions + three-step + 0f00-ba4 = cromwell-executions/three-step/0f00-ba4}}}
+   */
+  private def hostExecutionPath(workflow: WorkflowDescriptor): Path = Paths.get("cromwell-executions", workflow.name, workflow.id.toString)
+
+  /**
+   * {{{/root + cromwell-executions/three-step/0f00-ba4 + call-ps + stdout.out = /root/cromwell-executions/three-step/0f00-ba4/call-ps/stdout.out}}}
+   */
+  private def generateContainerPath(workflow: WorkflowDescriptor, directory: Path, file: Path): Path = {
+    Paths.get("/root", hostExecutionPath(workflow).toString, directory.toFile.getName, file.toFile.getName)
+  }
+
+  /**
+   * Converts paths of any call output `WdlFile`s from host to container paths.  Other `WdlValue`s are left unchanged.
+   */
+  override def localizeCallOutputs(workflow: WorkflowDescriptor, actualOutputs: Map[FullyQualifiedName, WdlValue]): Map[FullyQualifiedName, WdlValue] = {
+    actualOutputs map { case (fqn, value) =>
+      value match {
+        case WdlFile(hostPath) =>
+          fqn -> WdlFile(generateContainerPath(workflow, directory = hostPath.getParent, file = hostPath))
+        case x => fqn -> x
+      }
+    }
+  }
+
+  /**
+   * Given the specified workflow descriptor and inputs:
+   *
+   * <ol>
+   *   <li>Stage any WdlFiles into the workflow-inputs subdirectory of the workflow execution directory.</li>
+   *   <li>Return a version of the `WorkflowCoercedInputs` reflecting the container view of the path for
+   *   any `WdlFile`s.</li>
+   * </ol>
+   *
+   * Original input path: /could/be/anywhere/input.bam
+   * Host inputs path: $PWD/cromwell-executions/some-workflow-name/0f00-ba4/workflow-inputs/input.bam
+   * Container path: /root/cromwell-executions/some-workflow-name/0f00-ba4/workflow-inputs/input.bam
+   */
+  private def stageAndLocalizeWorkflowInputs(workflow: WorkflowDescriptor, actualInputs: WorkflowCoercedInputs): WorkflowCoercedInputs = {
+    actualInputs.map { case (fqn, wdlValue) =>
+      wdlValue match {
+        case WdlFile(originalInputPath) =>
+          // The fully qualified path to where the file should go in the host workflow-inputs directory.
+          val hostInputPath = Paths.get(hostExecutionPath(workflow).toFile.getAbsolutePath, "workflow-inputs", originalInputPath.toFile.getName)
+          Files.copy(originalInputPath, hostInputPath)
+          fqn -> WdlFile(generateContainerPath(workflow, directory = Paths.get("workflow-inputs"), file = originalInputPath))
+        case x => fqn -> x
+      }
+    }
+  }
```

Comment:
My understanding of this function is that for the JES backend this is essentially a no-op because all localization would have to happen at the Call level?  I could be wrong though